src/initialize_dae.jl

```diff
@@ -150,19 +174,26 @@ function _initialize_dae!(integrator, prob::ODEProblem, alg::ShampineCollocation
             out .-= tmp
             nothing
         end
-        r = nlsolve(nlequation!, u0, autodiff = isad ? :forward : :central,
-                    method = :newton)
-        integrator.u .= r.zero
+
+        nlsolve = default_nlsolve(alg.nlsolve, isinplace, u0, isad)
+
+        nlfunc = NonlinearFunction(nlequation!; jac = f.jac,
+                                   jac_prototype = f.jac_prototype)
```

Comment:
though it is true that this should be the same sparsity pattern, because by definition it's M/gamma - J